crates/holochain/src/conductor/conductor.rs

```diff
@@ -1595,25 +1603,15 @@ mod cell_impls {
         /// fully initialized and are registered with the kitsune network layer.
         /// Generally used to handle conductor interface requests.
         pub fn live_cell_ids(&self) -> HashSet<CellId> {
-            self.running_cell_ids(Some(CellStatusFilter::Joined))
+            self.running_cell_ids(|status| matches!(status, CellStatus::Joined))
         }
 
         /// List CellIds for Cells which match a status filter
-        pub fn running_cell_ids(&self, filter: Option<CellStatusFilter>) -> HashSet<CellId> {
+        pub fn running_cell_ids(&self, filter: impl Fn(&CellStatus) -> bool) -> HashSet<CellId> {
```

Comment:
I did this because in [at least one place](https://github.com/holochain/holochain/blob/82055ec47eead619ffe8ec1bf60ee6a5031e9ff0/crates/holochain/src/conductor/conductor/tests.rs#L1027), we need to match on `PendingJoin(_)`, i.e. that enum variant without regard to the inner reason, and it cleaned up that code, which was in danger of getting out of sync if another inner variant were added, since it was actually different filters for each possible variant and summing together the totals. So I opted for making this expressive enough to be able to get the needed value in a single filter call.